crates/l2/contracts/src/l1/CommonBridge.sol

```diff
@@ -194,32 +206,72 @@ contract CommonBridge is
     }
 
     /// @inheritdoc ICommonBridge
-    function sendToL2(SendValues calldata sendValues) public override whenNotPaused {
+    function sendToL2(
+        SendValues calldata sendValues
+    ) public override whenNotPaused {
         _sendToL2(_getSenderAlias(), sendValues);
     }
 
     /// @inheritdoc ICommonBridge
-    function deposit(address l2Recipient) public payable override whenNotPaused {
-        _deposit(l2Recipient);
-    }
+    function deposit(
+        uint256 _amount,
+        address l2Recipient
+    ) public payable override whenNotPaused {
+        uint256 value;
+
+        // Here we define value depending on whether the native token is ETH or an ERC20
+        if (NATIVE_TOKEN_L1_ADDRESS == address(0)) {
+            require(
+                msg.value > 0,
+                "CommonBridge: the native token is ETH, msg.value must be greater than zero"
+            );
+
+            require(
+                _amount == 0,
+                "CommonBridge: the native token is ETH, _amount must be zero"
+            );
+
+            value = msg.value;
+        } else {
+            require(
+                msg.value == 0,
+                "CommonBridge: the native token is an ERC20, msg.value must be zero"
+            );
+
+            require(
+                _amount > 0,
+                "CommonBridge: the native token is an ERC20, _amount must be greater than zero"
+            );
+
+            value = _amount;
+
+            // We lock the tokens in the bridge contract
+            IERC20(NATIVE_TOKEN_L1_ADDRESS).transferFrom(
+                msg.sender,
+                address(this),
+                value
+            );
+        }
+
+        deposits[ETH_TOKEN][ETH_TOKEN] += value;
```

Comment:
Shouldn't this be `NATIVE_TOKEN_L1_ADDRESS`? If not, it might be better to rename `ETH_TOKEN` as it can be confussing